lib/xapi-stdext-pervasives/pervasiveext.ml

```diff
@@ -15,59 +15,72 @@
  * Even if fct raises an exception, clean_f is applied
 *)
 
-let src = Logs.Src.create "pervasiveext" ~doc:"logs from Xapi_stdext_pervasives.Pervasiveext"
-
+let src =
+  Logs.Src.create "pervasiveext"
+    ~doc:"logs from Xapi_stdext_pervasives.Pervasiveext"
 
 let finally fct clean_f =
   let result =
-    try
-      fct ();
+    try fct ()
     with exn ->
-      Backtrace.is_important exn;
-      begin
-        (* We catch and log exceptions raised by clean_f to avoid shadowing
-           the original exception raised by fct *)
-        try
-          clean_f ();
+      Backtrace.is_important exn ;
+      ( try
+          (* We catch and log exceptions raised by clean_f to avoid shadowing
+             the original exception raised by fct *)
+          clean_f ()
         with cleanup_exn ->
-          Logs.warn ~src (fun m -> m "finally: Error while running cleanup after failure of main function: %s" (Printexc.to_string cleanup_exn));
-      end;
-      raise exn in
-  clean_f ();
-  result
-
-(* Those should go into the Opt module: *)
-
-let maybe_with_default d f v =
-  match v with None -> d | Some x -> f x
+          Logs.warn ~src (fun m ->
+              m
+                "finally: Error while running cleanup after failure of main \
+                 function: %s"
+                (Printexc.to_string cleanup_exn))
+      ) ;
+      raise exn
+  in
+  clean_f () ; result
 
-(** if v is not none, apply f on it and return some value else return none. *)
-let may f v = maybe_with_default None (fun x -> Some (f x)) v
+let maybe_with_default d f v = Option.fold ~none:d ~some:f v
 
-(** default value to d if v is none. *) 
-let default d v = maybe_with_default d (fun x -> x) v
+let may f v = Option.map f v
 
-(** apply f on v if not none *)
-let maybe f v = maybe_with_default () f v
+let default d v = Option.value ~default:d v
 
-(** if bool is false then we intercept and quiten any exception *)
-let reraise_if bool fct =
-  try fct () with exn -> if bool then raise exn else ()
+let maybe f v = Option.iter f v
 
 (** execute fct ignoring exceptions *)
 let ignore_exn fct = try fct () with _ -> ()
 
+(** if not bool ignore exceptions raised by fct () *)
+let reraise_if bool fct = if bool then fct () else ignore_exn fct
```

Comment:
Couldn't find any users for this function, so I've deprecated and changed the implementation to make its behaviour clearer.

Reply:
I find these functions of dubious value. It's fine if they are in the OCaml standard library but they are a bit too obscure for my taste. You can still define them locally when you need them a lot.

Reply:
I've deprecated `reraise_if` for removal. `ignore_exn` has some uses in xen-api:
```
rg ' ignore_exn ' -c
xen-api/ocaml/xapi/storage_migrate.ml:3
xen-api/ocaml/xapi/audit_log.ml:2
xen-api/ocaml/xapi/import.ml:2
xen-api/ocaml/database/redo_log.ml:2
xen-api/ocaml/database/block_device_io.ml:9
xen-api/ocaml/perftest/tests.ml:1
```

I'm ok deprecating it if we think it's worth changing these usages.

Reply:
I think we should deprecate `ignore_exn` as well - it's not even particularly useful since it doesn't allow `fct` to be polymorphic in its argument